### PR TITLE
adds docs for pantheon provider integration

### PR DESCRIPTION
**Pull request by tannerjfco**
## The Problem:
OP #369. All that fancy new pantheon functionality needs documentation!

## The Fix:
Here's some documentation! This PR is based on and revises the documentation @rickmanelius started in https://github.com/drud/ddev/issues/369#issuecomment-312716098. 

There are some sections I didn't include - namely the info on settings.php/wp-config.php (since it's no longer valid, and i'm not 100% on what is now valid), and Apache Solr, since I'm pretty sure we haven't attempted to use solr with a Pantheon site yet (I think this could be a follow-up item from the main pantheon work, personally). Otherwise, this should largely resemble what was being iterated in the Gdoc.

This will likely still require some changes, as development is still in progress.

## Related Issue Link(s):
#345 Pantheon Provider Plugin
#369 Pantheon Plugin Documentation

### CODE REVIEW

File: docs/users/providers/pantheon.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+ddev provides an integration with the [Pantheon Website Management Platform](https://pantheon.io/), which allows for Pantheon users to quickly and easily download and provision a site from Pantheon in a local ddev-managed environment.
+
+## Quick Start
+
+If you have ddev installed, and have an active Pantheon account with an active WordPress, Drupal 7, or Drupal 8 site, you can follow this quick start guide to spin up a pantheon site locally.
+
+1. Authenticate with Pantheon
+    a. Login to your Pantheon Dashboard, and [Generate a Machine Token](https://pantheon.io/docs/machine-tokens/) for ddev to use.
+    b. Run `ddev auth-pantheon <YOUR TOKEN>`
```
Comment (rickmanelius) [inline docs/users/providers/pantheon.md]: When you view the previewed version, the "a." and "b." are not showing up an indented child list. See preview here https://github.com/tannerjfco/ddev/blob/de1c39192703f1be2dfa41414bea64c04c99bf86/docs/users/providers/pantheon.md.

File: docs/users/providers/pantheon.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+2. Select the site you want to provision locally from your Pantheon Dashboard
```
Comment (rickmanelius) [inline docs/users/providers/pantheon.md]: Missing a period at the end.

File: docs/users/providers/pantheon.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+3. Select the environment you want to provision locally.
+3. Get a copy of the site code base from Pantheon. We recommend enabling the "Git Connection Mode" if not already enabled, and using `git clone` to checkout the code locally.
+4. Create a new backup of the site. This can be done by navigating to Backups->Backup Log->Create New Backup. _Note: this must be done every time you want the latest state of your Pantheon environment provisioned locally._
+5. Configure the local checkout for ddev
+    a. Navigate in your terminal to your checkout of the site codebase
+    b. Run `ddev config pantheon`
+    c. Follow the configuration prompts, selecting the Pantheon environment you want to sync from
```
Comment (rickmanelius) [inline docs/users/providers/pantheon.md]: We're seeing the ident problem here again, at least with the GitHub preview.

**Comment by rickmanelius** [comment]
Adding Brad as a reviewer given that this is based off a lot of his work.

File: docs/users/providers/pantheon.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Usage
+
+### Configuring a Pantheon Site for Imports
+
+After you copy your Pantheon site’s codebase locally, you can use `ddev config pantheon` to generate the necessary configuration files for ddev. In addition to the prompts the `ddev config` command provides for any new site, you will also be asked to specify which Pantheon environment you wish to pull your file and database assets from. These environments are usually "live", "test", or "dev". If you wish to later change the Pantheon environment you wish to sync from, you can do so by editing the `import.yaml` file in the `.ddev` directory for your site.
+
+### Imports
+
+Running `ddev pull` will connect to Pantheon through their API to find the latest versions of the database and files backups from the specified environment. If new versions are available on Pantheon, they are downloaded and stored in ~/.ddev. If the stored copies in ~/.ddev are the latest copies, ddev will use these cached copies instead of downloading them over again.
```
Comment (rfay) [inline docs/users/providers/pantheon.md]: Actually ~/.ddev/pantheon is where it is.

**Comment by rfay** [comment]
I added a commit dealing with my concerns, so this is approved by me. Thanks for your work on this Rick and Tanner!

**Comment by rfay** [comment]
I pulled it because I know Tanner would have wanted it pulled.

**Pull request merged**